### DISCUSSION TITLE
Could Outline view show the subroutines and sub-procedures like RDI?

### DISCUSSION THREAD

**Discussion by casweet72** [codefori/vscode-ibmi#2291] [Q&A]
Would it be possible to add  more functionality to the Outline view? Right now it only seems to list fields and data structures. It would be more useful for it to list the same information as the RDI outline view - like the Prototypes called, Subroutines and sub-procedures. It would also be nice to be able to collapse the list of fields in the view, in some programs this is a VERY long list.
The outline view in RDI is my main way of navigating through my code and is one of the bigger hurdles for me moving to VSCode.

**Comment by worksofliam** [comment]
The outline view will never be the same as RDi. This is intentional. The outline view is specifically for explicit definitions - files, structs, variables, procedures, subroutines, etc - not references.

You are able to navigate references in other ways. See our docs on RPGLE here: https://codefori.github.io/docs/extensions/rpgle/#content-assist

**Comment by casweet72** [comment]
Thanks!

**Comment by thomprl** [comment]
Yeah that is one thing I wish as well that it would look a little more like RDi.  I understand it's a different product but being able to set the references is handy.  I posted something here a while back.
https://github.com/orgs/codefori/discussions/1533
